pyuvdata/tests/__init__.py

```diff
@@ -5,19 +5,260 @@
+        if exc_type is None and exc_val is None and exc_tb is None:
+            if self.expected_warning is None:
+                assert len(self) == 0
+            else:
+                assert len(self) == len(self.expected_warning), (
+                    f"{len(self.expected_warning)} warnings expected, "
+                    f"{len(self)} warnings issued. The list of emitted warnings is: "
+                    f"{[each.message for each in self]}."
+                )
+
+                for warn_i, exp_warn in enumerate(self.expected_warning):
+                    if not any(issubclass(r.category, exp_warn) for r in self):
+                        __tracebackhide__ = True
+                        raise AssertionError(
+                            "DID NOT WARN. No warnings of type {} was emitted. "
+                            "The list of emitted warnings is: {}.".format(
+                                self.expected_warning, [each.message for each in self]
+                            )
+                        )
+                    elif self.match is not None:
+                        for record in self:
+                            if str(record.message).startswith(
+                                self.match[warn_i]
+                            ) or re.compile(self.match[warn_i]).search(
+                                str(record.message)
+                            ):
+                                if issubclass(
+                                    record.category, self.expected_warning[warn_i]
+                                ):
+                                    break
+                        else:
+                            raise AssertionError(
+                                f"No warnings of type {self.expected_warning[warn_i]} "
+                                f"matching ('{self.match[warn_i]}') was "
+                                "emitted. The list of emitted warnings is: "
+                                f"{[each.message for each in self]}."
+                            )
+
+
+def check_warnings(expected_warning, match=None, nwarnings=None, *args, **kwargs):
+    """
+    Assert that code raises a particular set of warnings, used as a context manager.
+
+    Similar to ``pytest.warns``, but allows for specifying multiple warnings.
+    It also better matches warning strings when the warning uses f-strings or
+    formating.
+
+    Note that unlike the older checkWarnings function, the warnings can be passed
+    in any order, they do not have to match the order the warnings are raised
+    in the code.
+
+    To assert that there are no warnings raised by some code, set `expected_warning`
+    to None (i.e. `with check_warnings(None):`)
+
+    Parameters
+    ----------
+    expected_warning : list of Warnings or Warning
+        List of expected warnings. If a single warning type or a length 1 list,
+        will be used for the type of all warnings.
+    match : list of str or str
```

Comment:
Yes, thanks. The regex part is copied exactly from `pytest.warns` and I think it will match things in the middle of the warning string. But I have found that it doesn't work properly for warnings that have fstrings or values inserted with a format call. I don't understand why it has that problem, but adding the `startswith` check has fixed it.